src/fsharp/InfoReader.fs

```diff
@@ -205,7 +225,7 @@ type HierarchyItem =
 
 /// An InfoReader is an object to help us read and cache infos. 
 /// We create one of these for each file we typecheck. 
-type InfoReader(g: TcGlobals, amap: Import.ImportMap) =
+type InfoReader(g: TcGlobals, amap: Import.ImportMap) as this =
```

Comment:
The `as this` should not be needed - make the method that consumes it into a member that has access to `this`

Reply:
Unfortunately, it's not that easy. I need access to 'this' because I want to perform a lazy operation of collecting runtime feature support info which requires looking at fields.